CandidateReranking.md

```diff
@@ -74,27 +72,27 @@ Entites returned from the SemanticSearchAPI.
 
 | Param    | Type | Description
 |----------|------|--------|
-| name | String | **Required**: Name of the keyword, e.g. "Java".
-| type | String | **Required**: Type of the keyword, e.g. "skill".
-| weight | Integer | **Required**: Weight of the keyword as an int.
+| name | String | **Optional**: Name of the keyword, e.g. "Java".
+| type | String | **Optional**: Type of the keyword, e.g. "skill".
+| weight | Integer | **Optional**: Weight of the keyword as an int.
 | relationships | Map<String, Entities[]> | **Optional**: Map of Strings(Semantic Enrichment Type) to a list of of entities. Valid list of Map key values are: `job_titles`, `occupations`, `skills`, `job_level`, `text_kernel_related_keywords`, `interesting_keywords`, `related_keywords`, `related_keywords_recruiter`, `related_keywords_jobseeker`, `raw_job_titles_jobseeker`, `raw_job_titles_recruiter`, `extracted_kewords`, `custom_keywords`, `related_search_terms`, and `job_titles_recruiter`.
```

Comment:
I don't think it looks good to use `Map<..>` (or even the word "Map" at all) when documenting a JSON API. "Map" is mostly a Java thing, so you're effectively leaking internal details/terminology from our service implementation into the documentation of its interface -- and its interface is JSON, not Java.

It might be sufficient to just use the word "dictionary" as the type. Or you may want to define a `Relationships` object further down and link to it here, like you did in some other places.

And my same comment from above about enum casing applies here (or is it just this way from SSAPI, so we can't change it here without breaking the convenience of "pass your SSAPI response directly to CRS?")

Also, typo on "extracted_kewords"

Reply:
Good point. I think I'll define it as a dictionary object and break the object into another section for further detail.